src/kernels/kernelFinder.base.ts

```diff
@@ -128,19 +130,22 @@ export abstract class BaseKernelFinder implements IKernelFinder {
     ): Promise<KernelConnectionMetadata[]> {
         this.startTimeForFetching = this.startTimeForFetching ?? new StopWatch();
 
-        try {
-            // Get both local and remote kernels.
-            const [localKernels, remoteKernels] = await Promise.all([
-                this.listLocalKernels(resource, cancelToken, useCache),
-                this.listRemoteKernels(resource, cancelToken, useCache)
-            ]);
+        // Get both local and remote kernels.
+        const [localKernels, remoteKernels] = await Promise.all([
+            this.listLocalKernels(resource, cancelToken, useCache).catch((ex) => {
+                traceError('Failed to get local kernels', ex);
+                return [];
+            }),
+            this.listRemoteKernels(resource, cancelToken, useCache).catch((ex) => {
+                traceError('Failed to get remote kernels', ex);
+                // When remote kernels fail, turn off remote
+                void this.serverUriStorage.setUri(Settings.JupyterServerLocalLaunch);
```

Comment:
This could be changed in the future to just remove the URI for the failed kernels. Since we only support one still, I force local if any remote failures occur.

Might be better here to check for a ECONNREFUSED? The way this is right now I think if there's any failure it will remove the URI from remote.

Reply:
But I figured Don was refactoring this anyway? Not sure if that's going to happen this month or not.

Reply:
I changed my mind. Made it check for 'econn' in the message.